File: server/src/main/com/broll/gainea/server/init/LobbyData.java
package com.broll.gainea.server.init;

import com.broll.gainea.net.NT_LobbySettings;
import com.broll.gainea.server.core.GameContainer;
import com.broll.networklib.server.impl.ILobbyData;

public class LobbyData implements ILobbyData {

    private final static int START_LOCATIONS_DEFAULT = 3;

    private final static int START_GOALS_DEFAULT = 3;

    private final static int POINT_LIMIT_DEFAULT = 0;

    private final static int ROUND_LIMIT_DEFAULT = 30;

    private final static int MONSTERS_PER_MAP = 10;

    private ExpansionSetting expansionSetting = ExpansionSetting.BASIC_GAME;

    private GoalTypes goalTypes = GoalTypes.ALL;

    private int startLocations = START_LOCATIONS_DEFAULT;

    private int startGoals = START_GOALS_DEFAULT;

    private int pointLimit = POINT_LIMIT_DEFAULT;

    private int monsterCount = MONSTERS_PER_MAP;

    private int roundLimit = ROUND_LIMIT_DEFAULT;

    private GameContainer game;

    private IGameStartListener gameStartListener;

    private boolean gameRoundsStarted = false;

    public void setGame(GameContainer game) {
        this.game = game;
    }

    public GameContainer getGame() {
        return game;
    }

    public void setExpansionSetting(ExpansionSetting expansionSetting) {
        this.expansionSetting = expansionSetting;
    }

    public void setGameStartListener(IGameStartListener gameStartListener) {
        this.gameStartListener = gameStartListener;
    }

    public IGameStartListener getGameStartListener() {
        return gameStartListener;
    }

    public ExpansionSetting getExpansionSetting() {
        return expansionSetting;
    }

    public int getStartGoals() {
        return startGoals;
    }

    public int getStartLocations() {
        return startLocations;
    }

    public void setStartGoals(int startGoals) {
        this.startGoals = startGoals;
    }

    public void setStartLocations(int startLocations) {
        this.startLocations = startLocations;
    }

    public GoalTypes getGoalTypes() {
        return goalTypes;
    }

    public int getPointLimit() {
        return pointLimit;
    }

    public void setGoalTypes(GoalTypes goalTypes) {
        this.goalTypes = goalTypes;
    }

    public void setPointLimit(int pointLimit) {
        this.pointLimit = pointLimit;
    }

    public void setMonsterCount(int monsterCount) {
        this.monsterCount = monsterCount;
    }

    public int getMonsterCount() {
        return monsterCount;
    }

    public void setRoundLimit(int roundLimit) {
        this.roundLimit = roundLimit;
    }

    public int getRoundLimit() {
        return roundLimit;
    }

    public boolean isGameRoundsStarted() {
        return gameRoundsStarted;
    }

    public void setGameRoundsStarted(boolean gameRoundsStarted) {
        this.gameRoundsStarted = gameRoundsStarted;
    }

    @Override
    public NT_LobbySettings nt() {
        NT_LobbySettings settings = new NT_LobbySettings();
        settings.expansionSetting = expansionSetting.ordinal();
        settings.startGoals = startGoals;
        settings.startLocations = startLocations;
        settings.pointLimit = pointLimit;
        settings.goalTypes = goalTypes.ordinal();
        settings.monsters = monsterCount;
        settings.roundLimit = roundLimit;
        return settings;
    }
}

File: server/src/main/com/broll/gainea/server/init/LobbyData.kt
package com.broll.gainea.server.init

import com.broll.gainea.net.NT_LobbySettings
import com.broll.gainea.server.core.Game
import com.broll.networklib.server.impl.ILobbyData

class LobbyData : ILobbyData {
    var expansionSetting = ExpansionSetting.BASIC_GAME
    var goalTypes = GoalTypes.ALL
    var startLocations = START_LOCATIONS_DEFAULT
    var startGoals = START_GOALS_DEFAULT
    var pointLimit = POINT_LIMIT_DEFAULT
    var monsterCount = MONSTERS_PER_MAP
    var roundLimit = ROUND_LIMIT_DEFAULT
    var game: Game? = null
    var gameStartListener: IGameStartListener? = null
    var gameRoundsStarted = false
    override fun nt(): NT_LobbySettings {
        val settings = NT_LobbySettings()
        settings.expansionSetting = expansionSetting.ordinal
        settings.startGoals = startGoals
        settings.startLocations = startLocations
        settings.pointLimit = pointLimit
        settings.goalTypes = goalTypes.ordinal
        settings.monsters = monsterCount
        settings.roundLimit = roundLimit
        return settings
    }

    companion object {
        private const val START_LOCATIONS_DEFAULT = 3
        private const val START_GOALS_DEFAULT = 3
        private const val POINT_LIMIT_DEFAULT = 0
        private const val ROUND_LIMIT_DEFAULT = 20
        private const val MONSTERS_PER_MAP = 10
    }
}
